patches/server/0756-Execute-chunk-tasks-mid-tick.patch

```diff
@@ -107,8 +107,28 @@ index 088334869cb62797a1e1d1bbb6187f03189d852d..c245c1f4611f7273c8da629f774e0c64
              return true;
          } else {
              if (this.haveTime()) {
+diff --git a/src/main/java/net/minecraft/server/level/ServerChunkCache.java b/src/main/java/net/minecraft/server/level/ServerChunkCache.java
+index 4f502f1c8a86a28e2cd6513516f3846ed74be891..b89596e3394698d91bb5720648916e01c251f52e 100644
+--- a/src/main/java/net/minecraft/server/level/ServerChunkCache.java
++++ b/src/main/java/net/minecraft/server/level/ServerChunkCache.java
+@@ -1023,6 +1023,7 @@ public class ServerChunkCache extends ChunkSource {
+                 iterator1 = shuffled.iterator();
+             }
+ 
++            int chunksTicked = 0; // Paper
+             try {
+             while (iterator1.hasNext()) {
+                 LevelChunk chunk1 = iterator1.next();
+@@ -1046,6 +1047,7 @@ public class ServerChunkCache extends ChunkSource {
+                         this.level.tickChunk(chunk1, k);
+                     }
+                 }
++                if ((chunksTicked++ & 1) == 0) net.minecraft.server.MinecraftServer.getServer().executeMidTickTasks(); // Paper
```

Comment:
Not sure when this changed (as in 1.17 it's as you have here), but at one point this call was made directly after the call to tickChunk (see https://github.com/Tuinity/Tuinity/blob/ver/1.16.5/patches/server/0015-Execute-chunk-tasks-mid-tick.patch#L234), so I think that should be fixed